Realm/RLMObjectStore.mm

```diff
@@ -517,15 +519,21 @@ id RLMGetObject(RLMRealm *realm, NSString *objectClassName, id key) {
         return nil;
     }
 
-    return RLMCreateObjectAccessor(realm, objectSchema, row);
+    return RLMCreateObjectAccessor(realm, objectSchema, (*objectSchema.table)[row]);
```

Comment:
Isn't the whole point of keeping the `RLMCreateObjectAccessor` variant that takes an index to be able to just pass in the index? IMO we could just remove that variant.